projects/dotcms-js/src/lib/core/site.service.ts

```diff
@@ -161,35 +161,38 @@ export class SiteService {
     }
 
     /**
-     * Request the current site
+     * Get the current site
      * @returns Observable<Site>
      * @memberof SiteService
      */
     getCurrentSite(): Observable<Site> {
-        if (this.selectedSite) {
-            return of(this.selectedSite);
-        } else {
-            return this.coreWebService
+        return merge(
```

Comment:
@alfredo-dotcms I updated this method that you created to be used [here](https://github.com/dotCMS/core-web/blob/issue-15620-api-link/src/app/view/components/_common/site-selector-field/site-selector-field.component.ts#L93) when we do QA on this issues can you please make sure that I didn't break anything by updating this method like this?